### PR TITLE
[15.0][MIG] base_exception: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT36446

### CODE REVIEW

File: base_exception/models/base_exception.py [hunk 1]
```diff
@@ -163,11 +159,11 @@ def _rule_eval(self, rule, rec):
...
                 _(
-                    "Error when evaluating the exception.rule rule:\n %s \n(%s)",
-                    rule.name,
-                    e,
+                    "Error when evaluating the exception.rule"
+                    " rule:\n %(rule_name)s \n(%(error)s)"
                 )
-            )
+                % {"rule_name": rule.name, "error": e}
+            ) from e
```
Comment (pedrobaeza) [inline base_exception/models/base_exception.py:166]: Why this `from e`?

File: base_exception/tests/test_base_exception.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def setUpClass(cls):
     @classmethod
     def tearDownClass(cls):
         cls.loader.restore_registry()
-        super().tearDownClass()
+        return super(TestBaseException, cls).tearDownClass()
```
Comment (pedrobaeza) [inline base_exception/tests/test_base_exception.py]: No need to add the arguments in `super`

File: base_exception/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2011 Raphaël Valyi, Renato Lima, Guewen Baconnier, Sodexis
+# Copyright 2017 Akretion (http://www.akretion.com)
+# Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# Copyright 2020 Hibou Corp.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Exception Rule",
+    "version": "15.0.2.1.0",
```
Comment (pedrobaeza) [inline base_exception/__manifest__.py]:
```suggestion
    "version": "15.0.1.0.0",
```

File: base_exception/models/base_exception.py (hunk 1 above)
Comment (cesar-tecnativa) [reply base_exception/models/base_exception.py:166]: pre-commit run message. when you raise an exception, raise the exception.

**Comment by cesar-tecnativa** [comment]
@chienandalu could you review this?

**Comment by cesar-tecnativa** [comment]
@pedrobaeza could you please review it?

**Comment by cesar-tecnativa** [comment]
@ernestotejeda please review

**Review by ernestotejeda — APPROVED** [review]

**Comment by cesar-tecnativa** [comment]
@LudLaf Please review

File: base_exception/__manifest__.py [hunk 4]
```diff
@@ -5,7 +5,8 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 {
     "name": "Exception Rule",
-    "version": "14.0.2.1.0",
+    "version": "15.0.2.1.0",
```
Comment (AaronHForgeFlow) [inline base_exception/__manifest__.py]: 15.0.1.0.0

**Review by AaronHForgeFlow — APPROVED** [review]
Code review

File: base_exception/__manifest__.py (hunk 4 above)
Comment (cesar-tecnativa) [reply base_exception/__manifest__.py]: Done

Comment (MiquelRForgeFlow) [reply base_exception/__manifest__.py]: Is not done yet :S

Comment (cesar-tecnativa) [reply base_exception/__manifest__.py]: Now it is 15.0.1.0.0

Comment (MiquelRForgeFlow) [reply base_exception/__manifest__.py]: Thanks

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Comment by AaronHForgeFlow** [comment]
BTW there was another PR open for this module (approved also). Are you aware of that?

**Comment by cesar-tecnativa** [comment]
> BTW there was another PR open for this module (approved also). Are you aware of that?

It was done because the previous one had problems with the tests. 
Sorry, I have just asked https://github.com/OCA/server-tools/pull/2319 to correct these issues.

**Comment by AaronHForgeFlow** [comment]
@cesar-tecnativa No worries, just asking, if this is PR is better perhaps we can ask the author to close that PR

**Comment by leemannd** [comment]
@cesar-tecnativa  @AaronHForgeFlow The best practice would be to make on the branch of the other PR

**Comment by pedrobaeza** [comment]
@leemannd no, as the other contributor in unresponsive and the migration is faulty, it's better to start from the scratch.

**Review by florentx — APPROVED** [review]
Looks good, please merge as it is a dependency for other add-ons, e.g. `sale_exception` in `OCA/sale-workflow`.

**Review by leemannd — APPROVED** [review]

File: base_exception/models/base_exception.py [hunk 5]
```diff
@@ -271,19 +265,6 @@ def _compute_exceptions_summary(self):
             else:
                 rec.exceptions_summary = False
 
-    def _popup_exceptions(self):
```
Comment (pedrobaeza) [inline base_exception/models/base_exception.py]: This method was used on `purchase_exception`, but it can be restored there, not here, while migrating it.

Comment (MiquelRForgeFlow) [reply base_exception/models/base_exception.py]: Also in `account_move_exception`. Please, accept https://github.com/OCA/server-tools/pull/2373.

**Pull request merged**